landlab/components/lateral_erosion/node_finder.py

```diff
@@ -0,0 +1,267 @@
+import math
+
+import numpy as np
+
+
+def angle_finder(grid, dn, cn, rn):
+    xcoord = grid.node_axis_coordinates(axis=0)
+    ycoord = grid.node_axis_coordinates(axis=1)
+    sl1 = (ycoord[cn] - ycoord[dn]) / (xcoord[cn] - xcoord[dn])
+    sl2 = (ycoord[rn] - ycoord[cn]) / (xcoord[rn] - xcoord[cn])
+    angle1 = math.degrees(math.atan(sl1))
+    angle2 = math.degrees(math.atan(sl2))
+    angle_diff = angle2 - angle1
+    angle_diff = abs(angle2 - angle1)
+    return angle_diff
+
+
+def forty_five_node(donor, i, receiver, neighbors, diag_neigh):
+    radcurv_angle = 0.67
+
+    lat_node = 0
+    # In Landlab 2019: diagonal list goes [NE, NW, SW, SE]. Node list are ordered as [E,N,W,S]
+    # if water flows SE-N OR if flow NE-S or E-NW or E-SW, erode west node
+    if (
```

Comment:
This chain of conditionals might be tricky to test. I'm wondering whether there might be a simpler method... (but it's not occurring to me what that might be;)

Reply:
agreed